samples/nextjs/src/components/styleguide/Styleguide-Tracking.tsx

```diff
@@ -1,7 +1,7 @@
 import React, { RefObject } from 'react';
 import { withSitecoreContext } from '@sitecore-jss/sitecore-jss-nextjs';
 import { trackingApi } from '@sitecore-jss/sitecore-jss-tracking';
-import { dataFetcher } from 'lib/data-fetcher';
+import { dataFetcher } from 'lib/data-fetcher-factory';
```

Comment:
Agree, I also was thinking about it. Seems like it's not a factory

Reply:
Not sure I follow on how it's different from our other "factories".

So a factory is supposed to define the purpose of a class or function, but decouple the consumers of the factory from the underlying implementation, right?

For example, we have `src\lib\dictionary-service-factory.ts`, which creates an instance of `RestDictionaryService`, which is a wrapper around `AxiosDataFetcher`. Here, the purpose of the factory is to abstract the initialization of the concrete class that will handle dictionary data fetching.

Similar, `src\lib\data-fetcher-factory.ts` returns a function which is a wrapper around the initialization of `AxiosDataFetcher`. Here, the purpose is also to abstract the concrete class that will handle data fetching. Even the comments say "Implements a data fetcher using Axios - replace with your favorite SSR-capable HTTP or fetch library if you like". So this function is defining what the 'data-fetcher' function should do, but encapsulates the implementation.

What do you guys think?

Reply:
What's missing for me in this case is any sort of _factory_. The other factories we've got in the sample define a factory class (i.e. `DictionaryServiceFactory`) which defines a "create" function, and this factory class is what is being exported.

Reply:
I think I get what you're saying. Seems like this goes back to our discussion of using functions directly vs wrapping functions in classes.

Option 1 (with class)
```
export class FooFactory {
  create() {
    return new FooService();
  }
}

export const fooFactory= new FooFactory();
```

Option 2 (just function)
```
export function fooFactory {
  return new FooService();
}
```

I see no benefit to wrapping the creation of FooService in a class and adding a create() function. Since `this` is never used in the class, what's the purpose of the class? Regardless, if it's confusing then I will change the filename back.

Reply:
I'm fine with a factory function (vs a factory class), but the problem is that the code in the `data-fetcher-factory.ts` isn't a factory function - it's just exporting a data fetcher implementation directly. It would need to be something like this to be considered a factory function.
```
export function createDataFetcher() {
  return function dataFetcher<ResponseType>(
    url: string,
    data?: unknown
  ): Promise<AxiosResponse<ResponseType>> {
    return new AxiosDataFetcher().fetch<ResponseType>(url, data);
  };
}

```
Your example above makes sense, since the `fooFactory` is our factory function.